yarn-project/aztec-rpc/src/account_state/account_state.ts

```diff
@@ -260,14 +260,14 @@ export class AccountState {
       : [];
 
     // 1 tx containing only 1 function invocation
-    const encryptedLogs = new TxL2Logs([executionResult.encryptedLogs]);
+    const encryptedLogs = new TxL2Logs(collectEncryptedLogs(executionResult));
```

Comment:
@benesjan can you check this is ok for getting encrypted logs from all nested calls?

Reply:
Currently the logs only contain the encrypted nullifier preimages/note preimages/note spending info. For it all to work it is necessary, that the order of the encrypted logs is the same as the order of commitments.

Your changes shouldn't break this commitment-log correspondence, no?

I am currently just pulling all the work together and I should probably write some tests specifically focused on the logs ordering.